volume-providers/vpc/vpcclient/models/volume_attachment.go

```diff
@@ -98,6 +98,13 @@ func (va *VolumeAttachment) ToVolumeAttachmentResponse() *provider.VolumeAttachm
 			devicepath = GTypeClassicDevicePrefix + va.Device.ID
 		}
 		varp.VolumeAttachmentRequest.VPCVolumeAttachment.DevicePath = devicepath
+
+		// TODO: This needs change, after device id for NG gets implemented
+	} else if va.Volume != nil && va.Volume.Generation == GTypeG2 {
+		if va.ID != "" && len(va.ID) >= 20 {
+			devicepath := GTypeG2DevicePrefix + va.ID[:20]
```

Comment:
is `va.ID` attachment ID then its fine else as per workaround it should be attachment ID?

Reply:
va.ID is attachment ID only (not volumeID).
will be changing deviceid as per riaas fixes